### PR TITLE
[WIP][ISSUE-3007][Improve] Introduce a premise utils to reduce conditions check and exceptions processing

### LINKED ISSUE #3007: [Improve] Introduce a utils to process conditions check and exceptions thrown.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

 Introduce a premise utils to reduce conditions check and exceptions processing.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengkezhou1**
## What changes were proposed in this pull request

Issue Number: close #3007 

## Brief change log

**Add a tool to unify pre-checking and throwing exceptions**

## Verifying this change

This change added tests and can be verified as follows:

- Added *Test to verify the change.*

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

File: streampark-common/src/main/java/org/apache/streampark/common/util/PremisesUtils.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+package org.apache.streampark.common.util;
+
+import org.apache.streampark.common.exception.AbstractApiException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Objects;
+
+/** Determine whether to throw an exception based on the check condition */
+public class PremisesUtils {
```
Comment (wolfboys) [inline streampark-common/src/main/java/org/apache/streampark/common/util/PremisesUtils.java]: good job

File: streampark-common/src/main/java/org/apache/streampark/common/domain/ResponseCode.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.streampark.console.base.domain;
+package org.apache.streampark.common.domain;
```
Comment (wolfboys) [inline streampark-common/src/main/java/org/apache/streampark/common/domain/ResponseCode.java]: hi, 

Thanks for your contribution! It's a great improvement. However, from my perspective, there are a issue:

The common layer is a globally module where we strive to include common functionalities that are unrelated to specific business logic. Besides providing a data platform based on Spring Boot, StreamPark is also a stream processing development framework. We have provided some Flink|spark rapid development APIs, allowing users to independently leverage the capabilities of the development framework. Looking at it this way, it is not reasonable to place platform-specific logic such as HTTP status codes and backend exception message display in the common layer. These are not common capabilities, and they are not needed by the development framework.

**Comment by wolfboys** [comment]
I really like this improvement and I'm looking forward to your next submit. 😊

**Comment by caicancai** [comment]
I will start reviewing this PR if no one reviews it

**Comment by wolfboys** [comment]
In the current code, the backend still returns error messages to the frontend as before. The key issue is that these error messages are hard-coded and not processed through [i18n](https://github.com/apache/incubator-streampark/issues/3520). You can check out this issue. My suggestion is that after the backend detects an error, it should return a business state code, and the message info for each status code can be handled by the frontend.

**Comment by caicancai** [comment]
@RocMarshal If you have time, can you help me with the review? Thank you.

**Comment by zhengkezhou1** [comment]
@RocMarshal @wolfboys I will replaced the hard-coded place in the project soon

**Comment by wolfboys** [comment]
hi:
There are some conflicts that need to be resolved first

**Comment by RocMarshal** [comment]
> In the current code, the backend still returns error messages to the frontend as before. The key issue is that these error messages are hard-coded and not processed through [i18n](https://github.com/apache/incubator-streampark/issues/3520). You can check out this issue. My suggestion is that after the backend detects an error, it should return a business state code, and the message info for each status code can be handled by the frontend.

+1 for me.

Hi @zzzk1 Thank you for the pr and sorry for the late reply.
thx for @wolfboys @caicancai comments.

How about 


- Only introduce a separate `XXXUtils` in console-service to perform a check with built-in exception binding
- And keep common related checks in `Utils`.

?

I'd like to more ideas about it~

**Comment by wolfboys** [comment]
> > In the current code, the backend still returns error messages to the frontend as before. The key issue is that these error messages are hard-coded and not processed through [i18n](https://github.com/apache/incubator-streampark/issues/3520). You can check out this issue. My suggestion is that after the backend detects an error, it should return a business state code, and the message info for each status code can be handled by the frontend.
> 
> +1 for me.
> 
> Hi @zzzk1 Thank you for the pr and sorry for the late reply. thx for @wolfboys @caicancai comments.
> 
> How about
> 
> * Only introduce a separate `XXXUtils` in console-service to perform a check with built-in exception binding
> * And keep common related checks in `Utils`.
> 
> ?
> 
> I'd like to more ideas about it~

* Only introduce a separate XXXUtils in console-service to perform a check with built-in exception binding
* And keep common related checks in Utils.

+1,  This is a good idea

**Pull request closed without merging**